### PR TITLE
DEV: Refactor ComposerMessages

### PR DESCRIPTION
**Pull request by CvX**
Extracted from #21075.

Conversion to a native class, `@action`, native getters, async/await, more isDestroying/isDestroyed checks, no unnecessary `scheduleOnce("afterRender")`, no unnecessary local variables, `||=` and `?.` operators, for-of loops.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/composer-messages.js [hunk 1]
```diff
@@ -31,100 +31,86 @@ export default Component.extend({
     this.appEvents.on("composer:find-similar", this, this._findSimilar);
     this.appEvents.on("composer-messages:close", this, this._closeTop);
     this.appEvents.on("composer-messages:create", this, this._create);
-    scheduleOnce("afterRender", this, this.reset);
-  },
+    this.reset();
+  }
 
   willDestroyElement() {
+    this._super(...arguments);
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/composer-messages.js:38]: Should be `super.willDestroyElement()`, right? *and likewise for other `this._super` references

**Review by davidtaylorhq — APPROVED** [review]

File: app/assets/javascripts/discourse/app/components/composer-messages.js (hunk 1 above)
Comment (CvX) [reply app/assets/javascripts/discourse/app/components/composer-messages.js:38]: yep yep yep!

File: app/assets/javascripts/discourse/app/components/composer-messages.js [hunk 2]
```diff
@@ -1,130 +1,118 @@
...
     this.appEvents.on("composer:find-similar", this, this._findSimilar);
     this.appEvents.on("composer-messages:close", this, this._closeTop);
     this.appEvents.on("composer-messages:create", this, this._create);
-    scheduleOnce("afterRender", this, this.reset);
-  },
+    this.reset();
+  }
 
   willDestroyElement() {
+    super.didInsertElement(...arguments);
```
Comment (davidtaylorhq) [inline app/assets/javascripts/discourse/app/components/composer-messages.js:39]:
```suggestion
    super.willDestroyElement(...arguments);
```
;)

Comment (CvX) [reply app/assets/javascripts/discourse/app/components/composer-messages.js:39]: lol

**Pull request merged**